scim/client.bal

```diff
@@ -0,0 +1,265 @@
+import ballerina/oauth2;
+import ballerina/http;
+import ballerina/io;
+
+# This client is used to connect to the SCIM2 APIs of Asgardeo for Inbound User Provisioning.
+@display {label: "SCIM Client Connector"}
+public class Client {
+    private final string orgName;
+    private final string clientId;
+    private final string clientSecret;
+    private final string[] scope;
+    final oauth2:ClientOAuth2Provider provider;
+    final http:Client clientEndpoint;
+    final map<string> headers;
+
+    # Initializes the SCIM client.
+    #
+    # + orgName - The name of the organization
+    # + clientId - The client ID of the application
+    # + clientSecret - The client secret of the application
+    # + scope - permitted scopes
+    public isolated function init(string orgName, string clientId, string clientSecret, string[] scope) returns error? {
+        self.orgName = orgName;
+        self.clientId = clientId;
+        self.clientSecret = clientSecret;
+        self.scope = scope;
+
+        self.provider = new ({
+            tokenUrl: "https://api.asgardeo.io/t/" + self.orgName + "/oauth2/token",
+            clientId: clientId,
+            clientSecret: clientSecret,
+            scopes: scope
+        });
+
+        self.clientEndpoint = check new ("https://api.asgardeo.io/t/" + self.orgName + "/scim2", {httpVersion: "1.1"});
+
+        string token = check self.provider.generateToken();
+        io:print(token);
+
+        self.headers = {
+            "accept": "application/scim+json",
+            "Authorization": "Bearer " + token
+        };
+    }
+
+    # Gets the list of users.
+    #
+    # + return - The list of users
+    @display {label: "Get Users"}
+    public isolated function getUsers() returns UserResponse|error {
+        json|error response = self.clientEndpoint->get("/Users", self.headers, json);
```

Comment:
repeat defining endpoints as constants for /Users as well. Check other applicable places.